skimage/filters/tests/test_edges.py

```diff
@@ -364,39 +364,138 @@ def test_laplace_mask():
     assert (np.all(result == 0))
 
 
-@testing.parametrize("grad_func", (
-    filters.prewitt_h, filters.sobel_h, filters.scharr_h
-))
+def test_farid_zeros():
+    """Farid on an array of all zeros."""
+    result = filters.farid(np.zeros((10, 10)), np.ones((10, 10), bool))
+    assert (np.all(result == 0))
+
+
+def test_farid_mask():
+    """Farid on a masked array should be zero."""
+    np.random.seed(0)
```

Comment:
I don't think we want to set seeds this way in our test suite.  What happens when we want to parallelize tests?